rust/src/storage/config.rs

```diff
@@ -230,16 +216,10 @@ impl ObjectStoreKind {
             }),
             #[cfg(feature = "gcs")]
             ObjectStoreKind::Google => {
-                let store = GoogleCloudStorageBuilder::new()
+                let store = GoogleCloudStorageBuilder::from_env()
```

Comment:
The integration tests were failing for GCP without this change. In the tests, we configure GCP through the `GOOGLE_SERVICE_ACCOUNT` environment variable, but instead this code was creating a GCS storage with default configuration.

Is there a reason we shouldn't be picking up the environment variables first? I assume that the storage options override them. @roeap

Reply:
That must have been a bug, or at least an inconsistency. I think for S3 and Azure we use `from_env` as well. That said., this is actually a point I struggled with a bit when writing this. Mainly for S3 and Azure where we have many different ways of authorising. One scenario in particular was troubling me. In case the environment contains enough information for the store to find a credential, and the storage options contain another complete set of config for a credential. In this case object_store will pick the credential based on the order they are checked in the builder, which does not necessarily give precedence to the one in the options, which I assume is what the user would want.

The initial approach was to try and create one only from the storage_options, and if we fail, load from environment and from options. TO be honest, I do not remember what the specific issue was why I abandoned this approach, but I do remember, that the special handling of `allow_http` is definitely related. In the end I guess, I though in scenarios where tables in multiple locations are managed people should be careful what to put in the env...

That said, anything in storage options will overwrite any info parsed from the environment.